strawberryfields/api_client.py

```diff
@@ -0,0 +1,546 @@
+        """
+        self.headers["Authorization"] = authentication_token
+
+    def join_path(self, path):
+        """
+        Joins a base url with an additional path (e.g. a resource name and ID)
+
+        Args:
+            path (str): A path to be joined with BASE_URL.
+
+        Returns:
+            str: A joined path.
+        """
+        return join_path(self.BASE_URL, path)
+
+    def get(self, path):
+        """
+        Sends a GET request to the provided path. Returns a response object.
+
+        Args:
+            path (str): A path to send the GET request to.
+
+        Returns:
+            requests.Response: A response object, or None if no response could be fetched from the
+            server.
+        """
+        try:
+            response = requests.get(url=self.join_path(path), headers=self.HEADERS)
+        except requests.exceptions.ConnectionError as e:
+            response = None
+            warnings.warn("Could not connect to server ({})".format(e))
+        return response
+
+    def post(self, path, payload):
+        """
+        Converts payload to a JSON string. Sends a POST request to the provided
+        path. Returns a response object.
+
+        Args:
+            path (str): A path to send the GET request to.
+            payload: A JSON serializable object to be sent to the server.
+
+        Returns:
+            requests.Response: A response object, or None if no response could be fetched from the
+            server.
+        """
+        # TODO: catch any exceptions from dumping JSON
+        data = json.dumps(payload)
+        try:
+            response = requests.post(url=self.join_path(path), headers=self.HEADERS, data=data)
+        except requests.exceptions.ConnectionError as e:
+            response = None
+            warnings.warn("Could not connect to server ({})".format(e))
+        return response
+
+
+class ResourceManager:
+    """
+    This class handles all interactions with APIClient by the resource.
+    """
+
+    http_status_code = None
+
+    def __init__(self, resource, client=None):
+        """
+        Initialize the manager with resource and client instances. A client
+        instance is used as a persistent HTTP communications object, and a
+        resource instance corresponds to a particular type of resource (e.g.
+        Job)
+        """
+        setattr(self, "resource", resource)
+        setattr(self, "client", client or APIClient())
+
+    def join_path(self, path):
+        """
+        Joins a resource base path with an additional path (e.g. an ID)
+        """
+        return join_path(self.resource.PATH, path)
+
+    def get(self, job_id):
+        """
+        Attempts to retrieve a particular record by sending a GET
+        request to the appropriate endpoint. If successful, the resource
+        object is populated with the data in the response.
+
+        Args:
+            job_id (int): The ID of an object to be retrieved.
```

Comment:
Does it make sense to have the name be somtething more general than `job_id`? elsewhere in the docstrings, you are more generic ("object", "record", etc.)

Reply:
Makes perfect sense, I forgot to rename this when I moved things to ResourceManager.